WordPress/src/main/java/org/wordpress/android/ui/plugins/PluginListFragment.java

```diff
@@ -103,23 +122,32 @@ void requestPlugins() {
     }
 
     void setListType(@NonNull PluginListType listType) {
-        // site plugins are retrieved all at once
-        if (listType == PluginListType.SITE) {
-            mCanLoadMore = false;
-        }
+        // site plugins are retrieved all at once so "load more" isn't necessary, search returns
+        // the first 50 best matches which we've decided is enough
+        mCanLoadMore = !(listType == PluginListType.SITE || listType == PluginListType.SEARCH);
 
-        if (!listType.equals(mListType)) {
-            mListType = listType;
-            getArguments().putSerializable(ARG_LIST_TYPE, mListType);
-            requestPlugins();
-        }
+        mListType = listType;
+        getArguments().putSerializable(ARG_LIST_TYPE, mListType);
+        requestPlugins();
     }
 
     PluginListType getListType() {
         return mListType;
     }
 
     private void setPlugins(@NonNull List<?> plugins) {
+        // preload .org plugins for site plugins
+        mWPOrgPluginsMap.clear();
```

Comment:
I am trying to understand the `setPlugins` method. If I am not mistaken, this can be called with both wporg and site plugins. It looks like we clear this map and then re-add them if the method is called with site plugins, but what happens if it's called with wporg ones. They don't seem to be handled at all.

Also, we have the `mSitePluginsMap`, shouldn't that map be also updated in this method?

Reply:
I admit it's a bit odd because I wanted the fragment/adapter to be able to handle both site plugins and WPorg plugins, so I'll clarify.

This code is the result of [this comment](https://github.com/wordpress-mobile/WordPress-Android/pull/7156#discussion_r165016406) on the previous plugins PR, which pointed out that I was hitting the DB in `onBindViewHolder()` to get the WPOrg plugin when a site plugin is being displayed. So, all this is doing is making sure we pre-load WPOrg plugins if we're passed site plugins, avoiding the DB hit in `onBindViewHolder()`.

The `mSitePluginsMap` is the reverse of that: making sure that when a list of WPOrg plugins is passed, we know which ones are also site plugins. That map only needs to be loaded when the fragment is created.

Reply:
I have been thinking a lot about the `mWPOrgPluginsMap` and I really don't think it should be something we utilize. I do agree with the underlying reason of this change, if we can avoid it we shouldn't hit the DB in `onBindViewHolder`. However, the wporg plugins data is one of the most dynamic ones we have in this feature. The data is fetched by multiple sources and they can return different details in some cases. Also, the plugin will not be in our DB initially, which means the map will need to be updated later (which we currently don't).

I do think we should do something about this and I have a few ideas I am exploring as part of a larger undertaking in ` issue/plugin-browser-view-model-pattern`. That's mostly an experiment and we might not end up merging it at all, but if any of avenues I pursue there proves fruitful, we can implement something similar in our master branch.

Just wanted to leave my thoughts on this, so if I propose a change soon, it doesn't come out of nowhere.